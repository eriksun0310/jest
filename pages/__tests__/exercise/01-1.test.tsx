import { render, screen } from "@testing-library/react";
import Button from "../../exercise/01";

// snapshot快照功能
test("Button displays hi", async () => {
  //gogo
  const { container, debug } = render(<Button />);
  // toMatchInlineSnapshot: 會將原本的Btn 快照起來,如果有更改過, 測試則不會過
  expect(container).toMatchInlineSnapshot(
  `
    <div>
      <button
        class="eb-bg-green-500 hover:eb-bg-green-700 eb-text-white eb-font-bold eb-py-2 eb-px-4 eb-rounded"
      >
        Hi
      </button>
    </div>
  `
  );
// toMatchSnapshot:產生新的檔案快照
expect(container).toMatchSnapshot()
});
