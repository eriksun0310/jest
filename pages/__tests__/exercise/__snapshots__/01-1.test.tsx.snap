// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button displays hi 2`] = `
<div>
  <button
    class="eb-bg-green-500 hover:eb-bg-green-700 eb-text-white eb-font-bold eb-py-2 eb-px-4 eb-rounded"
  >
    Hi
  </button>
</div>
`;
